File: minegame-database/src/main/java/top/nipuru/minegame/database/processor/PlayerTransactionHandler.java
package top.nipuru.minegame.database.processor;

import lombok.AllArgsConstructor;
import top.nipuru.minegame.common.message.database.PlayerTransactionRequest;
import top.nipuru.minegame.common.processor.RequestDispatcher;
import top.nipuru.minegame.database.DatabaseServer;

@AllArgsConstructor
public class PlayerTransactionHandler implements RequestDispatcher.Handler<PlayerTransactionRequest> {

    private final DatabaseServer server;

    @Override
    public void handle(RequestDispatcher.ResponseSender responseSender, PlayerTransactionRequest request) throws Exception {
        server.getPlayerDataManager().transaction(request);
        responseSender.sendResponse(true); // response
    }

    @Override
    public Class<PlayerTransactionRequest> interest() {
        return PlayerTransactionRequest.class;
    }
}

File: minegame-database/src/main/kotlin/top/nipuru/minegame/database/processor/PlayerTransactionHandler.kt
package top.nipuru.minegame.database.processor

import com.alipay.remoting.AsyncContext
import top.nipuru.minegame.common.message.database.PlayerTransactionRequest
import top.nipuru.minegame.common.processor.RequestDispatcher
import top.nipuru.minegame.database.player.PlayerDataManager

class PlayerTransactionHandler : RequestDispatcher.Handler<PlayerTransactionRequest> {

    override fun handle(asyncCtx: AsyncContext, request: PlayerTransactionRequest) {
        PlayerDataManager.transaction(request)
        asyncCtx.sendResponse(true) // response
    }

    override fun interest(): Class<PlayerTransactionRequest> {
        return PlayerTransactionRequest::class.java
    }
}
